File: elixir/domain_transfers.exs
# Usage: mix run domain_transfers.exs example.com 42
# where 42 is the domain transfer id

client = %Dnsimple.Client{
  access_token: Application.get_env(:dnsimple, :access_token),
  base_url: "https://api.sandbox.dnsimple.com/"
}

{:ok, response} = Dnsimple.Identity.whoami(client)
account_id = response.data.account.id

[name, transfer_id] = System.argv

IO.puts "Getting Transfer #{transfer_id} for domain #{name}"

IO.inspect Dnsimple.Registrar.get_domain_transfer(client, account_id, name, transfer_id)

File: node/domain_transfers.js
const { DNSimple, AuthenticationError } = require('dnsimple');

(async () => {
  const dnsimple = new DNSimple({
    baseUrl: 'https://api.sandbox.dnsimple.com',
    accessToken: process.env.TOKEN,
    userAgent: 'dnsimple-examples',
  });

  // Provide the domain name (e.g. example.com) as the first CLI argument...
  const name = process.argv[2];
  // ...and the transfer ID (e.g. 42) as the second CLI argument.
  const transferId = process.argv[3];

  try {
    const identity = await dnsimple.identity.whoami();
    const accountId = identity.data.account.id;
    const transfer = await dnsimple.registrar.getDomainTransfer(accountId, name, transferId);
    console.log(transfer);
  } catch (err) {
    if (err instanceof AuthenticationError) {
      console.error('Authentication error. Check your token is correct for the sandbox environment.');
    } else {
      console.error(err);
    }
  }
})();
